src/theory/quantifiers/sygus/sygus_interpol.cpp

```diff
@@ -40,25 +40,206 @@ SygusInterpol::SygusInterpol(LogicInfo logic) : d_logic(logic) {}
 void SygusInterpol::collectSymbols(const std::vector<Node>& axioms,
                                    const Node& conj)
 {
+  Trace("sygus-interpol-debug") << "Collect symbols..." << std::endl;
+  std::unordered_set<Node, NodeHashFunction> symsetAxioms;
+  std::unordered_set<Node, NodeHashFunction> symsetConj;
+  for (size_t i = 0, size = axioms.size(); i < size; i++)
+  {
+    expr::getSymbols(axioms[i], symsetAxioms);
+  }
+  expr::getSymbols(conj, symsetConj);
+  for (const Node& s : symsetAxioms)
+  {
+    d_syms.push_back(s);
+  }
+  for (const Node& s : symsetConj)
+  {
+    d_syms.push_back(s);
+  }
+  for (const auto& elem : symsetConj)
+  {
+    if (symsetAxioms.find(elem) != symsetAxioms.end())
+    {
+      d_symsetShared.insert(elem);
+    }
+  }
+  Trace("sygus-interpol-debug")
+      << "...finish, got " << d_syms.size() << " symbols in total. And "
+      << d_symsetShared.size() << " shared symbols." << std::endl;
 }
 
 void SygusInterpol::createVariables(bool needsShared)
 {
+  NodeManager* nm = NodeManager::currentNM();
+  for (const Node& s : d_syms)
+  {
+    TypeNode tn = s.getType();
+    if (tn.isConstructor() || tn.isSelector() || tn.isTester())
+    {
+      // datatype symbols should be considered interpreted symbols here, not
+      // (higher-order) variables.
+      continue;
+    }
+    // Notice that we allow for non-first class (e.g. function) variables here.
+    std::stringstream ss;
+    ss << s;
+    Node var = nm->mkBoundVar(tn);
+    d_vars.push_back(var);
+    Node vlv = nm->mkBoundVar(ss.str(), tn);
+    d_vlvs.push_back(vlv);
+    // TODO: bug fixed by hack (argument list for synthesis should be
+    // consistent)
+    if (!needsShared || d_symsetShared.find(s) != d_symsetShared.end())
+    {
+      d_varsShared.push_back(var);
+      d_vlvsShared.push_back(vlv);
+      d_varTypesShared.push_back(tn);
+    }
+  }
+  // make the sygus variable list
+  d_ibvlShared = nm->mkNode(kind::BOUND_VAR_LIST, d_vlvsShared);
+  Trace("sygus-interpol-debug") << "...finish" << std::endl;
 }
 
 std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> > getIncludeCons(
     const std::vector<Node>& assumptions, const Node& conclusion)
 {
+  NodeManager* nm = NodeManager::currentNM();
+  Assert(options::produceInterpols() != options::ProduceInterpols::NONE);
   std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> > result =
       std::map<TypeNode, std::unordered_set<Node, NodeHashFunction> >();
```

Comment:
Clarify why you need to initialize it like this (rather than just declaring).

Reply:
what do you mean?